src/opencl_common.c

```diff
@@ -2501,6 +2500,46 @@ void get_compute_capability(int sequential_id, unsigned int *major,
 	clGetDeviceInfo(devices[sequential_id],
 	                CL_DEVICE_COMPUTE_CAPABILITY_MINOR_NV,
 	                sizeof(cl_uint), minor, NULL);
+
+	if (!major) {
+/*
+ * Apple, VCL and some other environments don't expose CL_DEVICE_COMPUTE_CAPABILITY_MINOR_NV
+ * so we need this crap - which is incomplete, best effort matching.
+ * http://en.wikipedia.org/wiki/Comparison_of_Nvidia_graphics_processing_units
+ */
+		char dname[MAX_OCLINFO_STRING_LEN];
+
+		HANDLE_CLERROR(clGetDeviceInfo(devices[sequential_id],
+		                               CL_DEVICE_NAME,
+		                               sizeof(dname), dname, NULL),
+		               "clGetDeviceInfo for CL_DEVICE_NAME");
+
+		// Ampere 8.0
+		if (strstr(dname, "TITAN V") || strstr(dname, "RTX 30") ||
+		    (dname[0] == 'A' && dname[1] >= '1' && dname[1] <= '9'))
+			*major = 8;
+		// Volta 7.0, Turing 7.5
+		else if (strstr(dname, "TITAN V")) {
+			*major = 7;
+			if (strstr(dname, "RTX 20"))
+				*minor = 5;
+		}
+		// Pascal 6.x
+		else if (strstr(dname, "GT 10") || strstr(dname, "GTX 10"))
+			*major = 6;
+		// Maxwell 5.x
+		else if (strstr(dname, "GT 9") || strstr(dname, "GTX 9") || strstr(dname, "GTX TITAN X"))
+			*major = 5;
```

Comment:
BTW, I think there was also Titan XP or (was it Xp?) for Pascal. You could want to lookup its exact name and add to the Pascal block here.

Reply:
According to https://mc-stan.org/math/opencl_support.html it's Titan Xp but we should be safe using `strcasestr()` for it